src/common

```diff
@@ -229,23 +229,16 @@ OBJ += $(patsubst %.cc, %$(OBJEXT), $(filter %.cc, $(SRC)))
 
 # get version from git
 GIT_INFO = $(shell git describe --tags --always --dirty || echo "n/a")
-GIT_INFO_FILE = .release_info
+RELEASE_INFO = "\#define CBMC_VERSION \"$(CBMC_VERSION) ($(GIT_INFO))\""
+GIT_INFO_FILE = version.h
 
-CBMC_VERSION_FILES += $(filter %_parse_options$(OBJEXT), $(OBJ))
+$(GIT_INFO_FILE):
+	echo $(RELEASE_INFO) > $@
 
-ifeq ($(BUILD_ENV_),MSVC)
-$(CBMC_VERSION_FILES): CP_CXXFLAGS += /DCBMC_VERSION='"""$(CBMC_VERSION) ($(GIT_INFO))"""'
-else
-$(CBMC_VERSION_FILES): CP_CXXFLAGS += -DCBMC_VERSION="\"$(CBMC_VERSION) ($(GIT_INFO))\""
-endif
+$(filter %_parse_options$(OBJEXT), $(OBJ)): $(GIT_INFO_FILE)
 
 KNOWN_RELEASE_INFO = $(shell cat $(GIT_INFO_FILE) 2>/dev/null)
-ifneq ($(GIT_INFO), $(KNOWN_RELEASE_INFO))
-$(CBMC_VERSION_FILES): $(GIT_INFO_FILE)
-
-$(GIT_INFO_FILE):
-	echo $(GIT_INFO) > $@
-
+ifneq ($(RELEASE_INFO), $(KNOWN_RELEASE_INFO))
 .PHONY: $(GIT_INFO_FILE)
```

Comment:
Oh, I see - this is some sneaky logic to avoid updating the timestamp on the GIT_INFO_FILE if the version info hasn't changed - nice, but could you add a comment to explain that ?

Reply:
I'll add that comment, but the build failures on Travis and AppVeyor make me wonder whether this actually works as it should (it definitively does so locally).

Reply:
I had messed up quoting. Thus it did not actually work as supposed... I'll push an update addressing this once #2415 is merged.